docs/upgrades.md

```diff
@@ -0,0 +1,318 @@
+Upgrades and Installing Particular Versions
+===========================================
+
+When installing Pulp, there are 3 software components whose versions you should be mindful of:
+1. pulp_installer
+2. pulpcore
+3. 1 or more plugins
+
+If all you want to do is install Pulp and upgrade it to the very latest version in the future,
+you may not need to specify any versions. However, it is recommended to specify versions at the time
+of installation for 2 reasons
+* Handling a situation where not all your plugins compatible with the latest pulpcore version.
+* Doing updates/upgrade in the future in a controlled manner.
+
+There are 3 sets of **workflows** below. Each workflow supports installations and upgrades, the latter
+2 support updates and handling plugi:
+* ["Latest Version with Minimal Effort"](#latest-version-with-minimal-effort)
+* ["Specifying Minor Releases of Pulpcore and Plugins"](#specifying-minor-releases-of-pulpcore-and-plugins)
+* ["Specifying Exact Versions with Reproducibility"](#specifying-exact-versions-with-reproducibility)
+
+### Explanation of plugin version compatibility with Pulpcore
+
+Pulp 3 has a plugin architecture so that new content types and features can be added by the
+larger community. However, both pulpcore & plugins are installed by the installer using `pip`, which has limited
+dependency resolution. Plugins release at their own lifecycles. Thus in the worst case scenario, the
+latest release of plugin pulp_juicy could depend on the current minor version (e.g., 3.10.z) of
+pulpcore, while the latest release of pulp_sugary could depend on only older versions of pulpcore
+(e.g., 3.8.z or 3.9.z). But if pulp_juicy were to also support 3.9.z, then the installer can handle this
+use case with a workflow listed below.
+
+In order to avoid breaking multiple plugins for the sake of one plugin, and to avoid breaking existing
+installations, upgrading a plugin will not cause pulpcore to be updated as a dependency. Similarly,
+if there is an attempt to update a plugin to a version that is incompatible with pulpcore, the installer
+will fail and exit. The installer does a compatibility check early in the installation to prevent Pulp
+from being installed or upgraded to an incompatible set of versions.
+
+Thus you, yourself, must research plugin compatibility with the pulpcore version whenever you are
+installing one or more plugins. With each plugin release, the plugin compatibility is announced as
+part of the release announcement and included in the documentation for the specific plugin.
+
+
+### Latest Version with Minimal Effort:
+
+#### **Initial installation:**
+
+1. [Make sure](https://galaxy.ansible.com/pulp/pulp_installer) you are running the latest version of the installer, which
```

Comment:
do we also have documented the ansible/python version compatibility with each installer version?